_P164_MSswitch.ino

```diff
@@ -0,0 +1,169 @@
+#ifdef USES_P164
+//#######################################################################################################
+//#################################### Plugin 164: MSswitch #############################################
+//#######################################################################################################
+
+// Februray 3, 2019 initial release
+// 433MHZ Transmitters: http://www.rflink.nl/blog2/wiring
+// Known supported auto-learning switches are Nexa, Prove, Telldus & Jula Anslutt
+
+// List of commands:
+// (1) MSswitch,<G,1-16>,<ON/OFF>
+
+// Usage:
+// (1): Turn on switches in group (eg. MSswitch,G,on)
+
+#include <Socketswitch.h>
+
+// Download library from https://github.com/mortensalomon/Socket_Switch
+
+Socketswitch *Plugin_164_msswitch;
+
+#define PLUGIN_164
+#define PLUGIN_ID_164         99
+#define PLUGIN_NAME_164       "MS auto-learning Switch"
+#define PLUGIN_VALUENAME1_164 ""
+
+unsigned long SocketAddress;
+String log164;
```

Comment:
Do not declare this global. Better to declare it inside the function, then you know for sure the string will be destructed when the function has finished.
Try to keep the scope of variables as small as possible, then it is easier to predict their behavior.
A string that will remain allocated can grow and keep the memory allocated even when it is no longer needed, so it will keep its largest allocated size.